recipes-dom0/dom0-image-thin-initramfs/files/meta-xt-prod-extra/recipes-extended/guest-addons/files/guest_doma

```diff
@@ -0,0 +1,97 @@
+#!/bin/sh
+#
+# xencommons    Script to start DomA guest
+#
+# Author:       Andrii Anisov <andrii_anisov@epam.com>
+#
+# chkconfig: 2345 70 10
+# description: Starts and stops Android domain (DomA)
+### BEGIN INIT INFO
+# Provides:          guest_doma
+# Required-Start:    $guest_domd
+# Should-Start:
+# Required-Stop:     $guest_domd
+# Should-Stop:
+# Default-Start:     2 3 5
+# Default-Stop:      0 1 6
+# Short-Description: Start/stop DomA
+# Description:       Starts and stops Android domain (DomA)
+### END INIT INFO
+
+DOMAIN_NAME="DomA"
+SCRIPT_NAME="start_doma.sh"
+SCRIPT_PATH="/xt/scripts"
+SCRIPT_ARGS="displbe bridge"
+
+# not running in Xen dom0 or domU
+if ! test -d /proc/xen ; then
+	exit 0
+fi
+
+do_start () {
+    echo "Starting ${DOMAIN_NAME}"
+    if xl domid ${DOMAIN_NAME} &> /dev/null; then
+	echo "${DOMAIN_NAME} is already running"
+        exit 0
+    fi
+
+    if pidof -x start_doma.sh > /dev/null ; then
+	echo "${DOMAIN_NAME} start is waiting for backends"
+	exit 0
+    fi
+
+    ${SCRIPT_PATH}/${SCRIPT_NAME} ${SCRIPT_ARGS} &
+    echo "${DOMAIN_NAME} is scheduled to run"
+}
+
+do_stop () {
+    echo "Stopping ${DOMAIN_NAME}"
+
+    killall ${SCRIPT_NAME} &> /dev/null
+
+    if ! xl domid ${DOMAIN_NAME} &> /dev/null ; then
+	echo "${DOMAIN_NAME} is not running"
+        exit 0
+    fi
+    xl destroy ${DOMAIN_NAME}
+}
+
+do_status () {
+    if ! xl domid ${DOMAIN_NAME} &> /dev/null ; then
+	echo "${DOMAIN_NAME} is not running"
+
+        if pidof -x ${SCRIPT_NAME} > /dev/null ; then
+	    echo "${DOMAIN_NAME} start is waiting for backends"
+	fi
```

Comment:
Again problem with spaces.
Also, I already have seen this code. Maybe it is better to make separate function that checks state?

Reply:
In the other place is an exit from inside of "if" block. Another function would only add a code complexity.